feat(cli): add config

aaravm opened the pull request:
Here, most of the code is copied #38, only getting config is added. Please merge this after #38.
Also, cargo.toml is the same as #38 except, anyhow crate is added.

<!-- Generated by sourcery-ai[bot]: start summary -->

## Summary by Sourcery

Add a new CLI tool for managing tasks with subcommands for creating, listing, getting, checking status, and canceling tasks. Include functionality to read configuration from a JSON file and update the documentation with usage examples.

New Features:
- Introduce a CLI tool with subcommands for managing tasks, including 'create', 'list', 'get', 'status', and 'cancel'.

Enhancements:
- Add functionality to read configuration from a JSON file and use it within the CLI tool.

Documentation:
- Provide examples in the documentation for using the CLI tool with various subcommands.

<!-- Generated by sourcery-ai[bot]: end summary -->

pavelnikonorov commented on cli/src/main.rs:
```diff
@@ -0,0 +1,303 @@
+    let mut contents = String::new();
+    file.read_to_string(&mut contents)?;
+    
+    let json_value: Value = serde_json::from_str(&contents)?;
```
as the SDK/CLI works with JSON data in APIs it handles, it might also be the optimal way to store config as JSON. Not sure if it's best for user experience. For the reference: AWS CLI – INI format,  Google Cloud CLI – text-based files with key-value pairs, Azure CLI – JSON format, IBM Cloud CLI – YAML format.

aaravm replied:
I initially had tried implementing it like that only (of course, it is the neatest way to solve the problem), but the problem is that I think some data types in Configuration Struct are not deserializable, for eg the Url crate. If you know of a workaround, or are able to do this, please do tell me, and I will look into it again.

uniqueg replied:
For the specific problem, maybe this? https://docs.rs/url_serde/latest/url_serde/

As for how to store the config: JSON sounds good to me. Or why not use YAML? It's easier to manipulate, is a superset of JSON and can almost always be directly converted to JSON.

aaravm replied:
Changed

pavelnikonorov commented on cli/src/main.rs:
```diff
@@ -0,0 +1,303 @@
+                    let funnel_url = ensure_funnel_running().await;
+                    config.set_base_path(&funnel_url);
+                }
+                match TES::new(&config).await {
```
@aaravm, I agree that a TES instance could be created in the beginning of `Some(("tes", sub)) => {` section, same with config load. It will significantly simplify this code

aaravm replied:
I have refactored the code

pavelnikonorov reviewed (changes requested):
I've highlighted two places with a potential for code simplifaction: config load and deduplication of TES-init code

aaravm commented on cli/src/main.rs:
```diff
@@ -0,0 +1,303 @@
+
+fn load_configuration() -> Configuration {
+    let config_file_path = dirs::home_dir().map(|path| path.join(".config/config.json"));
+    if let Some(path) = config_file_path {
```
Oh, this is a much simpler implementation! Changed

aaravm commented:
Hey @pavelnikonorov @uniqueg, I have refactored the code to separate config files, as suggested by you guys. I also changed the type of input in list in #38. These were the 2 suggestions given by @pavelnikonorov. Please merge this and #38.

pavelnikonorov commented:
@uniqueg 

1. Rebased this branch on the previous work in https://github.com/elixir-cloud-aai/ga4gh-sdk/pull/38 to avoid further merge conflicts.
2. refactored the configuration reader by integrating it as a `Configuration struct` method.
3. created an issue for multi-profile configuration https://github.com/elixir-cloud-aai/ga4gh-sdk/issues/47

I think it's good to merge.

pavelnikonorov commented:
All proposed changes have been addressed in the PR, and I will proceed with merging it as discussed with @uniqueg.
